File: VNCore/EnvironmentVariables.cs
﻿using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Text;

namespace VNCore
{
    public static class EnvironmentVariables
    {
        public const string NovelExtension = "vnxml", PackedNovelExtension = "vnzip";
        static string DataDirectory = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.ApplicationData), "VisNoveller");
        public static string NovelsDirectory = Path.Combine(DataDirectory, "Novels");
        public static string ConfigurationFile = Path.Combine(DataDirectory, "Config.xml");
    }
}

File: VNCore.Java/EnvironmentVariables.java
import java.nio.file.Paths;

public class EnvironmentVariables {
    public void setDataDirectory(String _dataDirectory) {
        dataDirectory = _dataDirectory;
        novelsDirectory = Paths.get(dataDirectory, "Novels").toString();
        configurationFile = Paths.get(dataDirectory, "Config.xml").toString();
    }

    public static final String NOVEL_EXTENSION = "vnxml",
            PACKED_NOVEL_EXTENSION = "vnzip";
    static String dataDirectory = Paths.get("").toAbsolutePath().toString();
    static String novelsDirectory = Paths.get(dataDirectory, "Novels").toString();
    static String configurationFile = Paths.get(dataDirectory, "Config.xml").toString();
}